packages/browser-destinations/src/destinations/sprig-web/trackEvent/__tests__/index.test.ts

```diff
@@ -0,0 +1,48 @@
+import { Analytics, Context } from '@segment/analytics-next'
+import sprigWebDestination, { destination } from '../../index'
+import { Subscription } from '../../../../lib/browser-destinations'
+
+const subscriptions: Subscription[] = [
+  {
+    partnerAction: 'trackEvent',
+    name: 'Track Event',
+    enabled: true,
+    subscribe: 'type = "track"',
+    mapping: {
+      name: {
+        '@path': '$.name'
+      },
+      anonymousId: {
+        '@path': '$.anonymousId'
+      }
+    }
+  }
+]
+
+describe('trackEvent', () => {
+  test('it maps event parameters correctly to track function ', async () => {
+    const [trackEvent] = await sprigWebDestination({
+      envId: 'testEnvId',
+      subscriptions
+    })
+
+    destination.actions.trackEvent.perform = jest.fn()
+    jest.spyOn(destination.actions.trackEvent, 'perform')
+    await trackEvent.load(Context.system(), {} as Analytics)
+
+    await trackEvent.track?.(
+      new Context({
+        type: 'track',
+        name: 'Button Clicked',
+        anonymousId: 'anonymous-id-0'
+      })
+    )
+
+    expect(destination.actions.trackEvent.perform).toHaveBeenCalledWith(
```

Comment:
throughout, i'm mocking the actual handler because I'm not able to use `jest` to mock/track calls on window.Sprig